longtailstorelib/gcsstore.go

```diff
@@ -120,16 +120,17 @@ func (blobObject *gcsBlobObject) Read() ([]byte, error) {
 }
 
 func (blobObject *gcsBlobObject) LockWriteVersion() (bool, error) {
-	blobObject.writeCondition = &storage.Conditions{DoesNotExist: true}
 	objAttrs, err := blobObject.objHandle.Attrs(blobObject.ctx)
+	if err == nil {
+		blobObject.writeCondition = &storage.Conditions{MetagenerationMatch: objAttrs.Metageneration, DoesNotExist: false}
+
+		return true, nil
+	}
 	if err == storage.ErrObjectNotExist {
+		blobObject.writeCondition = &storage.Conditions{DoesNotExist: true}
 		return false, nil
 	}
```

Comment:
Nit: The following structure is preferred in Go:

```go
if err == storage.ErrObjectNotExists {
   // Handle object not existing
} else if err != nil {
   // Handle genuine error
}

// Handle object existing
```

Reply:
fixed